extensions/wikia/AuthModal/js/AuthModal.js

```diff
@@ -17,12 +19,31 @@ define('AuthModal', ['jquery', 'wikia.window'], function ($, window) {
 		$(window).on('message.authPopUpWindow', function (event) {
 			var e = event.originalEvent;
 
-			if (typeof e.data !== 'undefined' && e.data.isUserAuthorized) {
+			if (typeof e.data === 'undefined') {
+				return;
+			}
+
+			if (e.data.isUserAuthorized) {
 				close();
 				if (typeof onAuthSuccess === 'function') {
 					onAuthSuccess();
 				}
 			}
+
+			if (e.data.beforeunload && !closeTrackTimeoutId) {
+				// to avoid tracking 'close' action whenever the window is reloaded;
+				closeTrackTimeoutId = setTimeout(function () {
+					clearTimeout(closeTrackTimeoutId);
```

Comment:
Why do we need to clear timeout? While it's executed it's already cleared I believe.